### ISSUE TITLE
[a11y] have a way to test no text gets clipped in non scroll direction when enlarged

### ISSUE THREAD

**Issue by chunhtai** [flutter/flutter#170705]
### Use case

according to https://www.w3.org/WAI/WCAG21/Understanding/reflow

enlarging the ui should not cause text to wrap around in non-scrolling direction. 

There is also no reason that a text should be clipped without wrap around or ellipsis. We should have a way to test that no text should be clipped by its parent box unless the directly that clipped is scrollable.

### Proposal

have a way to test that no text should be clipped by its parent box unless the directly that clipped is scrollable.

**Comment by chunhtai** [comment]
also can be used to test text spacing shouldn't end up clipping the text
https://www.w3.org/WAI/WCAG21/Understanding/text-spacing
